pgjdbc/src/main/java/org/postgresql/util/PGServiceFile.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (c) 2018, PostgreSQL Global Development Group
+ * See the LICENSE file in the project root for more information.
+ */
+
+package org.postgresql.util;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.text.MessageFormat;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Scanner;
+
+/*
+ * See https://www.postgresql.org/docs/current/static/libpq-pgservice.html.
+ */
+public class PGServiceFile {
+  public static Map load(String service) throws Exception {
+    String filename = findPath();
+    PGServiceFile file = new PGServiceFile();
+    Scanner in = new Scanner(new FileInputStream(filename), "US-ASCII");
+    try {
+      file.parse(in);
+    } finally {
+      in.close();
+    }
+    return file.getService(service);
+  }
+
+  private static String findPath() {
+    String filename = System.getProperty("org.postgresql.pgservicefile");
+    if (filename == null) {
+      filename = System.getenv().get("PGSERVICEFILE");
+    }
+    if (filename == null) {
+      filename = System.getProperty("user.home") + File.separator + ".pgservices.conf";
+    }
+    return filename;
+  }
+
+  private Map<String, Map<String, String>> sections;
+
+  public PGServiceFile() {
+    sections = new HashMap();
+  }
+
+  public void parse(Scanner in) throws Exception {
+    String line = null;
+    int lineNum = 0;
+    String sectionName = null;
+    Map<String, String> section = null;
+
+    while (in.hasNextLine()) {
+      line = in.nextLine();
+      lineNum++;
+      line = line.replaceAll("^\\s+", "");
+      if (line.isEmpty() || line.startsWith("#")) {
+        continue;
+      } else if (line.startsWith("[")) {
+        if (!line.endsWith("]")) {
+          String msg = MessageFormat.format("Error in service file line {0}: missing ].", lineNum);
+          throw new Exception(msg);
+        }
+        sectionName = line.substring(1, line.length() - 1);
+        section = new HashMap();
+        sections.put(sectionName, section);
+      } else if (section == null) {
+        String msg = MessageFormat.format("Error in service file line {0}: not in section.", lineNum);
+        throw new Exception(msg);
+      } else {
+        String[] segment = line.split("=");
```

Comment:
This should be `line.split("=", 2)` as it's okay to have `"` in the value. All that matters is there's at least one so that the name / value can be identified.

That way `username=test=123` becomes key `username` with value `test=123`.

Reply:
Agree, Fixed.